### PR TITLE
[firebase_ml_vision] Bug fix: if confidence is 1 or 0 it is decoded as int

### PR DESCRIPTION
**Pull request by PolinaGo**
## Description

When `confidence` (used by `ImageLabel` and `TextContainer`) is 1 or 0 and the value is passed over the channel, on the Dart side it is decoded as `int`, which leads to the following error:
```
Unhandled Exception: type 'int' is not a subtype of type 'double'
```

This fix forces the value to be interpreted as `double`.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_ml_vision/test/firebase_ml_vision_test.dart [hunk 1]
```diff
@@ -795,7 +795,7 @@ void main() {
                 'languageCode': 'cd',
               }
             ],
-            'confidence': 0.1,
+            'confidence': 0,
```
Comment (kroikie) [inline packages/firebase_ml_vision/test/firebase_ml_vision_test.dart:798]: Should we still have tests of confidence being `0.1`?

File: packages/firebase_ml_vision/test/image_labeler_test.dart [hunk 2]
```diff
@@ -33,7 +33,7 @@ void main() {
       test('processImage', () async {
         final List<dynamic> labelData = <dynamic>[
           <dynamic, dynamic>{
-            'confidence': 0.6,
+            'confidence': 1,
```
Comment (kroikie) [inline packages/firebase_ml_vision/test/image_labeler_test.dart:36]: Should we still have a test for `0.6` confidence?

File: packages/firebase_ml_vision/test/firebase_ml_vision_test.dart (hunk 1 above)
Comment (PolinaGo) [reply packages/firebase_ml_vision/test/firebase_ml_vision_test.dart]: Validity of confidence was checked multiple times with various non-corner case values.
However, it is probably better to add new test cases instead of modifying old ones to avoid introducing more errors by accident. 

Confidence here is now changed to the original value.

File: packages/firebase_ml_vision/test/image_labeler_test.dart (hunk 2 above)
Comment (PolinaGo) [reply packages/firebase_ml_vision/test/image_labeler_test.dart]: Same as above, 

Confidence here is now changed to the original value.

File: packages/firebase_ml_vision/test/firebase_ml_vision_test.dart (hunk 1 above)
Comment (kroikie) [reply packages/firebase_ml_vision/test/firebase_ml_vision_test.dart]: Thanks for updating, feel free to remove tests that you believe are unnecessary or redundant in a follow up PR.
